### PR TITLE
Mechanically enforces minimum age requirements for jobs

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
Pretty much a port of https://github.com/Baystation12/Baystation12/pull/27197

- [x] test the thing

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!wip

File: code/modules/mob/living/carbon/human/species/species.dm [hunk 1]
```diff
@@ -11,6 +11,7 @@
 	var/short_name                                       // Shortened form of the name, for code use. Must be exactly 3 letter long, and all lowercase
 	var/blurb = "A completely nondescript species."      // A brief lore summary for use in the chargen screen.
 	var/bodytype
+	var/respects_age_requirements = TRUE                 // Does this species respect the job minimum age requirements? For diona, IPC, vaurca.
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/carbon/human/species/species.dm]: Not so sure about this

**Comment by NonQueueingMatt** [comment]
!review

File: code/modules/client/preference_setup/occupation/occupation.dm [hunk 2]
```diff
@@ -182,6 +184,9 @@
 		else if (ban_reason)
 			dat += "<del>[rank]</del></td><td><b> \[<a href='?src=\ref[user.client];view_jobban=\ref[rank];'>BANNED</a>]</b></td></tr>"
 			continue
+		else if((character_age < job.minimum_character_age) && species.respects_age_requirements)
```
Comment (Karolis2011) [inline code/modules/client/preference_setup/occupation/occupation.dm]:
```suggestion
		else if(species.respects_age_requirements && (character_age < job.minimum_character_age))
```

File: code/modules/client/preference_setup/occupation/occupation.dm [hunk 3]
```diff
@@ -182,6 +184,9 @@
...
+			dat += "<del>[rank]</del></td><td> \[MIN CHAR AGE: [job.minimum_character_age]]</td></tr>"
```
Comment (Karolis2011) [inline code/modules/client/preference_setup/occupation/occupation.dm]:
```suggestion
			dat += "<del>[rank]</del></td><td> \[MINIMUM AGE: [job.minimum_character_age]]</td></tr>"
```

**Comment by NonQueueingMatt** [comment]
!wip

**Comment by NonQueueingMatt** [comment]
!review

**Comment by alsoandanswer** [comment]
...why, though? I haven't seen any problems, and even then are rarely enforced by admins unless for good reason.

Feels kind of handholdy. In real life, there are people still hired much younger than the age requirement e.g. them taking an apprenticeship or taking over someone.

I feel that we can trust our playerbase enough.

**Comment by NonQueueingMatt** [comment]
It's against the rules to join as someone younger than the minimum ages on the wiki.

The admins handle cases like this everyday. You just don't see it. Why note people if we can mechanically prevent it instesd?

**Comment by alsoandanswer** [comment]
Yo have you checked whether saved characters are retroactively updated to match the minumum age requirements

**Comment by NonQueueingMatt** [comment]
No need. They won't be able to join if their age is fucked.

File: code/modules/mob/living/carbon/human/species/species.dm (hunk 1 above)
Comment (Arrow768) [reply code/modules/mob/living/carbon/human/species/species.dm]: IT might be better to use a species flag for that.

Comment (NonQueueingMatt) [reply code/modules/mob/living/carbon/human/species/species.dm]: done

**Comment by NonQueueingMatt** [comment]
I don't know whether to feel offended or motivated

**Pull request merged**